### PR TITLE
[MIG] partner_event: Migration to 10.0

### PR DESCRIPTION
**Pull request by chienandalu**
- Minor migration tweaks

cc @Tecnativa

### CODE REVIEW

File: partner_event/__manifest__.py [hunk 1]
```diff
@@ -1,43 +1,26 @@
...
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# -*- coding: utf-8 -*-
+# Copyright 2014 Tecnativa S.L. - Pedro M. Baeza
+# Copyright 2015 Tecnativa S.L. - Javier Iniesta
+# Copyright 2016 Tecnativa S.L. - Antonio Espinosa
+# Copyright 2016 Tecnativa S.L. - Vicent Cubells
+# Copyright 2016 Tecnativa S.L. - David Vidal
```
Comment (cubells) [inline partner_event/__manifest__.py]: Change year please

File: partner_event/__manifest__.py [hunk 2]
```diff
@@ -1,43 +1,26 @@
...
     'name': 'Link partner to events',
-    'version': '8.0.1.1.0',
+    'version': '10.0.1.0.0',
     'category': 'Marketing',
-    'author': 'Serv. Tecnol. Avanzados - Pedro M. Baeza, '
-              'Antiun Ingeniería S.L., '
+    'author': 'Tecnativa,'
               'Odoo Community Association (OCA)',
-    'website': 'http://www.serviciosbaeza.com, http://www.antiun.com',
+    'website': 'http://www.tecnativa.com',
```
Comment (cubells) [inline partner_event/__manifest__.py]: s/http/https/g

File: partner_event/models/event_event.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+# © 2014 Tecnativa S.L. - Pedro M. Baeza
+# © 2015 Tecnativa S.L. - Javier Iniesta
+# © 2016 Tecnativa S.L. - Antonio Espinosa
+# © 2016 Tecnativa S.L. - Vicent Cubells
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields
```
Comment (cubells) [inline partner_event/models/event_event.py]: Reorder imports alphabetically please

File: partner_event/models/event_registration.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+# © 2014 Tecnativa S.L. - Pedro M. Baeza
+# © 2015 Tecnativa S.L. - Javier Iniesta
+# © 2016 Tecnativa S.L. - Antonio Espinosa
+# © 2016 Tecnativa S.L. - Vicent Cubells
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, api, fields
```
Comment (cubells) [inline partner_event/models/event_registration.py]: Same as above

File: partner_event/models/res_partner.py [hunk 5]
```diff
@@ -1,9 +1,11 @@
...
-##############################################################################
+# -*- coding: utf-8 -*-
+# © 2014 Tecnativa S.L. - Pedro M. Baeza
+# © 2015 Tecnativa S.L. - Javier Iniesta
+# © 2016 Tecnativa S.L. - Antonio Espinosa
+# © 2016 Tecnativa S.L. - Vicent Cubells
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from openerp import models, fields, api
+from odoo import api, fields, models
```
Comment (cubells) [inline partner_event/models/res_partner.py:8]: Same as above

File: partner_event/models/res_partner.py [hunk 6]
```diff
@@ -12,20 +14,42 @@ class ResPartner(models.Model):
     registrations = fields.One2many(
         string="Event registrations",
         comodel_name='event.registration', inverse_name="partner_id")
+    event_count = fields.Integer(
+        string='Events',
+        compute='_compute_event_count',
+        help="Count of events with confirmed registrations.",
+    )
     registration_count = fields.Integer(
         string='Event registrations number', compute='_count_registration',
```
Comment (cubells) [inline partner_event/models/res_partner.py]: compute method must start by _compute

File: partner_event/models/res_partner.py [hunk 7]
```diff
@@ -12,20 +14,42 @@ class ResPartner(models.Model):
...
         store=True)
     attended_registration_count = fields.Integer(
         string='Event attended registrations number',
         compute='_count_attended_registration', store=True)
```
Comment (cubells) [inline partner_event/models/res_partner.py]: Same as above

File: partner_event/wizard/res_partner_register_event.py [hunk 8]
```diff
@@ -1,21 +1,23 @@
...
-##############################################################################
+# -*- coding: utf-8 -*-
+# © 2014 Tecnativa S.L. - Pedro M. Baeza
+# © 2015 Tecnativa S.L. - Javier Iniesta
+# © 2016 Tecnativa S.L. - Antonio Espinosa
+# © 2016 Tecnativa S.L. - Vicent Cubells
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from openerp import models, fields, api
+from odoo import models, fields, api
```
Comment (cubells) [inline partner_event/wizard/res_partner_register_event.py]: Reorder alphabetically

**Review by cubells — CHANGES_REQUESTED** [review]
Fast code review

**Review by yajo — APPROVED** [review]
Not sure why codecov is ❌, but code seems OK.

**Review by luismontalba — APPROVED** [review]
Tested on runbot

File: partner_event/models/res_partner.py (hunk 5 above)
Comment (chienandalu) [reply partner_event/models/res_partner.py:8]: Fixed, but it doesn't outdate

Comment (yajo) [reply partner_event/models/res_partner.py:8]: I'd say you forgot this line. Can you double-check?

Comment (chienandalu) [reply partner_event/models/res_partner.py:8]: Well, it's sorted alphabetically as it's shown in the diff

Comment (yajo) [reply partner_event/models/res_partner.py:8]: ah right

**Comment by yajo** [comment]
@cubells are you happy now?

**Review by cubells — APPROVED** [review]
:+1: 
LGTM now

File: partner_event/README.rst [hunk 9]
```diff
@@ -17,33 +17,54 @@ It also includes:
...
-If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
-`here <https://github.com/OCA/event/issues/new?body=module:%20partner_event%0Aversion:%208.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
+In case of trouble, please check there if your issue has already been
+reported. If you spotted it first, help us smashing it by providing a detailed
+and welcomed feedback.
+
+Known Issues / Roadmap
+======================
 
+* Remove some TODOs if https://github.com/odoo/odoo/pull/12997 is merged.
```
Comment (yajo) [inline partner_event/README.rst]: https://github.com/odoo/odoo/pull/12997 was merged already, could you attend the TODOs please?

**Review by yajo — CHANGES_REQUESTED** [review]
One very little thing to do 😊

File: partner_event/__init__.py [hunk 10]
```diff
@@ -1,9 +1,4 @@
 # -*- coding: utf-8 -*-
-# © 2014 Tecnativa S.L. - Pedro M. Baeza
-# © 2015 Tecnativa S.L. - Javier Iniesta
-# © 2016 Tecnativa S.L. - Antonio Espinosa
-# © 2016 Tecnativa S.L. - Vicent Cubells
-# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
Comment (dreispt) [inline partner_event/__init__.py]: Not blocking, but I rather leave this as it was.

File: partner_event/models/res_partner.py [hunk 11]
```diff
@@ -20,15 +20,15 @@ class ResPartner(models.Model):
...
         store=True)
     attended_registration_count = fields.Integer(
         string='Event attended registrations number',
-        compute='_count_attended_registration', store=True)
+        compute='_compute_attended_registrations', store=True)
 
     @api.multi
     @api.depends('registrations')
-    def _count_registration(self):
+    def _compute_registrations(self):
```
Comment (dreispt) [inline partner_event/models/res_partner.py]: Not an issue, but I would have chosen `_compute_count_registration`

**Review by dreispt — APPROVED** [review]
LGTM

File: partner_event/models/res_partner.py [hunk 12]
```diff
@@ -20,15 +20,16 @@ class ResPartner(models.Model):
...
+	compute='_compute_count_registration',
         store=True)
     attended_registration_count = fields.Integer(
         string='Event attended registrations number',
         compute='_compute_attended_registrations', store=True)
 
     @api.multi
     @api.depends('registrations')
-    def _compute_registrations(self):
+    def _compute_count_registration(self):
```
Comment (dreispt) [inline partner_event/models/res_partner.py]: If you're willing to do this rename, there are more similar cases in this PR that could also be changed.

Comment (chienandalu) [reply partner_event/models/res_partner.py]: I'm opened to your proposals :)

Comment (yajo) [reply partner_event/models/res_partner.py]: I think @dreispt means that the compute name should [follow the guidelines](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#models): `_compute_<fieldname>`.

File: partner_event/views/event_registration_view.xml [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+
+<record id="view_registration_search" model="ir.ui.view">
+    <field name="name">Search attended registrations</field>
+    <field name="model">event.registration</field>
+    <field name="inherit_id" ref="event.view_registration_search"/>
+    <field name="arch" type="xml">
+        <!-- TODO Change expr if https://github.com/odoo/odoo/pull/12997 is merged -->
```
Comment (yajo) [inline partner_event/views/event_registration_view.xml]: @chienandalu You removed the comment from the roadmap, but I meant that you actually fix this please 😉

**Review by yajo — CHANGES_REQUESTED** [review]
Oops, I think I didn't explain myself fine with the last review...

File: partner_event/views/event_registration_view.xml (hunk 13 above)
Comment (chienandalu) [reply partner_event/views/event_registration_view.xml]: It doesn't apply to this version. The same PR could be made though

Comment (yajo) [reply partner_event/views/event_registration_view.xml]: Yikes, true. Then restore the Roadmap note please.

Comment (yajo) [reply partner_event/views/event_registration_view.xml]: Oh no, wait, it is fixed in 10.0: https://github.com/odoo/odoo/commit/f6692139da96bf25ed0699157e53254373607415.

File: partner_event/views/event_registration_view.xml [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+     Copyright 2016 Vicent Cubells <vicent.cubells@tecnativa.com>
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+<record id="view_registration_search" model="ir.ui.view">
+    <field name="name">Search attended registrations</field>
+    <field name="model">event.registration</field>
+    <field name="inherit_id" ref="event.view_registration_search"/>
+    <field name="arch" type="xml">
+        <filter name="open" position="after">
```
Comment (yajo) [inline partner_event/views/event_registration_view.xml]: Seems like [they changed it to `expected`](https://github.com/odoo/odoo/blob/10.0/addons/event/views/event_views.xml#L391).

**Review by yajo — CHANGES_REQUESTED** [review]

**Comment by chienandalu** [comment]
All green at last :smiley:

**Review by yajo — APPROVED** [review]
Perfect! Please do the final squash of your commits to merge 😊

File: partner_event/models/event_registration.py [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+    @api.model
+    def create(self, vals):
+        if not vals.get('partner_id') and vals.get('email'):
+            partner_model = self.env['res.partner']
+            event_model = self.env['event.event']
+            partner_id = False
+            # Look for a partner with that email
+            email = vals.get('email').replace('%', '').replace('_', '\\_')
+            partners = partner_model.search(
+                [('email', '=ilike', email)])
```
Comment (pedrobaeza) [inline partner_event/models/event_registration.py]: Use `limit=1` instead and you will have directly a `partner`

File: partner_event/models/res_partner.py [hunk 16]
```diff
@@ -12,20 +14,43 @@ class ResPartner(models.Model):
...
+    def _compute_registration_count(self):
+        for partner in self:
+            partner.registration_count = len(partner.registrations)
 
-    @api.one
+    @api.multi
+    def _compute_event_count(self):
+        for partner in self:
+            partner.event_count = len(
+                self.env["event.registration"].search([
```
Comment (pedrobaeza) [inline partner_event/models/res_partner.py:40]: Use `search_count` instead for having directly the count

File: partner_event/views/res_partner_view.xml [hunk 17]
```diff
@@ -1,32 +1,33 @@
...
+<odoo>
 
 <record id="act_partner_registration" model="ir.actions.act_window">
     <field name="res_model">event.registration</field>
     <field name="view_type">form</field>
     <field name="name">Registrations</field>
     <field name="view_mode">tree,form,calendar,graph</field>
-    <field name="context">{'search_default_partner_id': active_id, 'default_partner_id': active_id}</field>
+    <field name="domain">[("partner_id", "child_of", active_ids)]</field>
+    <field name="context">{'default_partner_id': active_id, 'search_default_expected': True}</field>
```
Comment (pedrobaeza) [inline partner_event/views/res_partner_view.xml]: Why do you remove the search_default_partner_id key?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: partner_event/views/res_partner_view.xml (hunk 17 above)
Comment (yajo) [reply partner_event/views/res_partner_view.xml]: Is it possible that `default_partner_id` implies that?

Comment (chienandalu) [reply partner_event/views/res_partner_view.xml]: I don't really know. That change comes from @cubells commit previous to this migration

File: partner_event/models/res_partner.py (hunk 16 above)
Comment (chienandalu) [reply partner_event/models/res_partner.py:40]: The problem here is that a partner can have multiple registrations for the same event. Thus the `mapped("event_id")` is used on the obtained records to obtain the unique count.

File: partner_event/views/res_partner_view.xml (hunk 17 above)
Comment (pedrobaeza) [reply partner_event/views/res_partner_view.xml]: No, they are different things. I fear that the filter activated with this context key has been replaced with a specific domain like `('id', 'in', ids)`. I prefer this form as it allows to refresh and it performs better.

Comment (chienandalu) [reply partner_event/views/res_partner_view.xml]: As I've seen inside the code, the filter seems to be still valid. So I recover it.

File: partner_event/models/res_partner.py (hunk 16 above)
Comment (pedrobaeza) [reply partner_event/models/res_partner.py:40]: OK, I see...

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**